### DISCUSSION TITLE
Embedding a dashboard using the Embedded SDK results in the chart POST request failing

### DISCUSSION THREAD

**Discussion by Anshu1294** [apache/superset#24606] [Q&A / Help]
Currently trying to embed superset charts for our POC, and after resolving many different issues, I'm completely stuck on the issue described here. I've tried many different things to identify what is the cause of the issue, without any success.

When you use the embedded sdk to embed a chart, a bunch of different APIs are called, to get charts, dashboards, datasets, but i believe the API call below returns the actual data that the chart is supposed to display:
(this is the POST call as seen in the swagger): 

![image](https://github.com/apache/superset/assets/17434863/7e7ace62-7388-4c5e-a7a1-6021532aa6de)

**https://<superset instance>/api/v1/chart/data?form_data=%7B%22slice_id%22%3A16%7D&dashboard_id=15&force**

When i loaded the dashboard in superset, i see the exact same API call as above. The only difference there is that it works with a correct response.

The one thing i've been able to confirm is that if i essentially copy the request as a curl from the network tab where i'm trying to embed the superset dashboard and replace the 'x-guesttoken' header and use a session cookie instead, the request works fine.

Here is the error i get back when a superset guesttoken is used:

{
    "message": "Error: 400 Unrecognized name: string at [56:8]\n\nLocation: EU\nJob ID: 5b9f388e-6772-49c9-9dfe-3db1ef2b9668\n"
}

![image](https://github.com/apache/superset/assets/17434863/54da003f-d1c8-4e6a-89a7-35457b8e5957)

For more context, our data is contained within BigQuery, and the suspicion is that user impersonation when running GBQ requests is the reason why we get this error. Superset logs couldn't tell me much either.

I'm wondering if the fact that a guest token is used is the problem here. And if so, what can I do to circumvent this issue. I'm happy to provide any more information. Appreciate the community response.

**Comment by jimmydavid** [comment]
Hi man,

Can you solve the issue?

Im stuck in the same method but i ger a: Error: ORA-00936: missing expression
Help: https://docs.oracle.com/error-help/db/ora-00936/

I paste my steps:

`Hi, I am getting an error in an ebedded dashboard using @superset-ui/embedded-sdk in a React app:

In chrome at network tab I see passed ok the following methods,

api/v1/guesttoken
api/v1/embedded/dashboard_id?uiConfig=1
api/v1/me/roles
api/v1/dashboard/idDashboard
api/v1/dashboard/idDashboard/charts
api/v1/dashboard/idDashboard/datasets
But fail in this step
api/v1/chart/data?form_data.... with Payload sending and headers X-Csrftoken and X-Guestoken
I got a 400 Bad request with this response
{
"message": "Error: ORA-00936: missing expression\nHelp: https://docs.oracle.com/error-help/db/ora-00936/"
}
The embedded get the titles and format same as viewed in Superset without data only showing this error:

Error: ORA-00936: missing expression
Help: https://docs.oracle.com/error-help/db/ora-00936/

I comment that in Superset the dashboard is loading succesfully.

Someone with same error.

Thanks in advanced`

**Reply by silmerusse** [reply]
I resolved my problem according to https://github.com/apache/superset/issues/29954#issuecomment-2292440452
